lib/ExecutionContext.js

```diff
@@ -173,6 +173,22 @@ class ExecutionContext {
     });
     return createJSHandle(this, response.objects);
   }
+
+  /**
+   * @param {Puppeteer.ElementHandle} elementHandle
+   * @return {Promise<Puppeteer.ElementHandle>}
+   */
+  async _adoptElementHandle(elementHandle) {
+    assert(elementHandle.executionContext() !== this, 'Cannot adopt handle that already belongs to this execution context');
```

Comment:
Fine for now. But in the case of a future public version we should probably `return elementHandle` here instead of asserting.

Reply:
Public API will be either on DOMWorld or on ElementHandle, so we'll have an opportunity to think about it